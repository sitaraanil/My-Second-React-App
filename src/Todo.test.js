import React from "react";
import ReactDOM from "react-dom";
import Todo from "./Todo";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("Todo", () => {
    const description = "New Todo";
    const mockRemoveTodo = jest.fn();
    const component = shallow(<Todo description={description} removeTodo={mockRemoveTodo} />);

    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Todo />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders and matches our snapshot", () => {
        const component = renderer.create(<Todo description="Yo" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders a Todo component", () => {
        expect(component.contains(<div className="Todo" />));
    });

    it("contains the description", () => {
        expect(component.text()).toContain(description);
    });

    it("marks the Todo as done", () => {
        component.find("button.MarkDone").simulate("click");
        expect(component.state("done")).toEqual(true);
    });
});