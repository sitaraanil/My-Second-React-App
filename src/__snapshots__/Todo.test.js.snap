// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders and matches our snapshot 1`] = `
<div
  className="Todo"
>
  Yo
  <br />
  <hr />
  <button
    onClick={[Function]}
  >
    Mark as done
  </button>
  <button
    onClick={[Function]}
  >
    Remove Me
  </button>
</div>
`;
